### PR TITLE
Pass scalac arguments as file

### LINKED ISSUE #2553: `--server=false` fails for a massive number of files in 10k range
**Version(s)**
1.0.5

**Describe the bug**
When compiling a project with a massive number of files, around 10k and more, and the `--server=false` flag, it fails with:
```
java.io.IOException: Cannot run program "/opt/homebrew/Cellar/openjdk/21.0.1/libexec/openjdk.jdk/Contents/Home/bin/java" (in directory "/Users/pprazak/repos/besom/
.out/codegen/azure/5.55.0"): error=7, Argument list too long
  java.base@17.0.6/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
  java.base@17.0.6/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
  scala.build.internal.Runner$.run0(Runner.scala:91)
  scala.build.internal.Runner$.run(Runner.scala:48)
...
java.io.IOException: error=7, Argument list too long
  java.base@17.0.6/java.lang.ProcessImpl.forkAndExec(ProcessImpl.java)
  java.base@17.0.6/java.lang.ProcessImpl.<init>(ProcessImpl.java:314)
  java.base@17.0.6/java.lang.ProcessImpl.start(ProcessImpl.java:244)
  java.base@17.0.6/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
  java.base@17.0.6/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
  scala.build.internal.Runner$.run0(Runner.scala:91)
  scala.build.internal.Runner$.run(Runner.scala:48)
  scala.build.internal.Runner$.runJvm(Runner.scala:172)
```

**To Reproduce**
Git clone https://github.com/VirtusLab/besom

```
just get-schema gcp 7.0.0
just generate-provider-sdk gcp 7.0.0
scala-cli --power publish local --server=false .out/codegen/gcp/7.0.0 --suppress-experimental-feature-warning
```

**Expected behaviour**
`scala-cli` should compile an arbitrary number of files.

Thanks to @WojciechMazur we know the solution, pass a file instead of args:
> @\<file\>  A text file containing compiler arguments (options and source files).

### PR DESCRIPTION
**Pull request by MaciejG604**
Fixes #2553

### CODE REVIEW

**Review by Gedochao — APPROVED** [review]

File: modules/build/src/main/scala/scala/build/Build.scala [hunk 1]
```diff
@@ -948,6 +950,8 @@ object Build {
 
     val project = Project(
       directory = inputs.workspace / Constants.workspaceDirName,
+      argsFilePath =
+        projectRootDir(inputs.workspace, inputs.projectName) / Constants.scalacArgumentsFileName,
```
Comment (tgodzik) [inline modules/build/src/main/scala/scala/build/Build.scala:954]: Where does it end up in `.scala-build` ?

**Review by tgodzik — APPROVED** [review]
LGTM! Just one question.

File: modules/build/src/main/scala/scala/build/Build.scala (hunk 1 above)
Comment (MaciejG604) [reply modules/build/src/main/scala/scala/build/Build.scala:954]: In `.scala-build/project-name-12345-HASH-12345/scalac.args.txt`, so it's kept together with the hashed contents of the compilation process, it should not get accessed by some other call to compile with different configuration

Comment (tgodzik) [reply modules/build/src/main/scala/scala/build/Build.scala:954]: Cool, thanks!

File: build.sc [hunk 2]
```diff
@@ -431,6 +431,8 @@ trait Core extends ScalaCliSbtModule with ScalaCliPublishModule with HasTests
          |  def workspaceDirName = "$workspaceDirName"
          |  def projectFileName = "$projectFileName"
          |  def jvmPropertiesFileName = "$jvmPropertiesFileName"
+         |  def scalacArgumentsFileName = "scalac.args.txt"
+         |  def maxScalacArgumentsCount = 5000
```
Comment (pawelprazak) [inline build.sc:435]: unfortunately I'm  affraid this will be too low for our use case, sice we codegen tens of thousands of files in some modules

Comment (MaciejG604) [reply build.sc:435]: This is just a limit above which arguments are passed as a `@<file>`

So this should fix the problem, check out the tests added.

Comment (pawelprazak) [reply build.sc:435]: Sorry, I've misread the implementation. Thank you for the fix! :)

File: modules/build/src/test/scala/scala/build/tests/BuildTests.scala [hunk 3]
```diff
@@ -930,4 +931,24 @@ abstract class BuildTests(server: Boolean) extends TestUtil.ScalaCliBuildSuite {
...
   }
+
+  for (options <- Seq(defaultOptions, defaultScala3Options))
+    test(s"compile 12k sources for Scala ${options.scalaOptions.scalaVersion.get.asString}") {
+      val mainInput = os.rel / "main.sc" ->
+        """//> using jvm 11
+          |println("Hello from big build")
+          |""".stripMargin
+
+      val additionalInputs = 1 to 12000 map { i =>
```
Comment (pawelprazak) [inline modules/build/src/test/scala/scala/build/tests/BuildTests.scala:942]: 🥇

**Pull request merged**